### ISSUE TITLE
l10n_it_delivery_note sequenza multi anno

### ISSUE THREAD

**Issue by sergiocorato** [OCA/l10n-italy#3094]
## Module
l10n_it_delivery_note

## Describe the bug
la sequenza del dn non considera la data

## To Reproduce
**Affected versions**:
- [x] 12.0: #3087 
- [x] 14.0: #3099 
- [ ] 16.0: #3175 superseed by #4543

1. creare una dn in data diversa dall'anno in corso
2. alla conferma prende la sequenza dell'anno in corso invece di quella corretta

**Expected behavior**
viene presa la sequenza dalla date range corretta sulla data del dn 

**Additional context**

**Comment by francesco-ooops** [comment]
@odooNextev questa è presente sulla 16?

**Comment by odooNextev** [comment]
> @odooNextev questa è presente sulla 16?

No, manca la correzione nella sequenza ed il test (un file con lo stesso nome è presente, ma è il mio per le multicompany): https://github.com/OCA/l10n-italy/blob/16.0/l10n_it_delivery_note/models/stock_delivery_note.py#L508

**Comment by francesco-ooops** [comment]
@sergiocorato puoi riaprire la issue?

**Comment by SirAionTech** [comment]
@sergiocorato `16.0` c'è https://github.com/OCA/l10n-italy/pull/3175

**Comment by TheMule71** [comment]
#4543 risolve anche questa issue
